pkg/odo/cli/catalog/describe/component.go

```diff
@@ -0,0 +1,207 @@
+package describe
+
+import (
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/openshift/odo/pkg/catalog"
+	"github.com/openshift/odo/pkg/devfile"
+	"github.com/openshift/odo/pkg/devfile/parser"
+	"github.com/openshift/odo/pkg/devfile/parser/data/common"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/machineoutput"
+	"github.com/openshift/odo/pkg/odo/genericclioptions"
+	"github.com/openshift/odo/pkg/odo/util/experimental"
+	"github.com/openshift/odo/pkg/odo/util/pushtarget"
+	pkgUtil "github.com/openshift/odo/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"k8s.io/klog"
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
+)
+
+const componentRecommendedCommandName = "component"
+
+var (
+	componentExample = ktemplates.Examples(`  # Describe a component
+    %[1]s nodejs`)
+
+	componentLongDesc = ktemplates.LongDesc(`Describe a component type.
+This describes the component and its associated starter projects.
+`)
+)
+
+// DescribeComponentOptions encapsulates the options for the odo catalog describe component command
+type DescribeComponentOptions struct {
+	// name of the component to describe, from command arguments
+	componentName string
+	// if devfile components with name that matches arg[0]
+	devfileComponents []catalog.DevfileComponentType
+	// if componentName is a classic/odov1 component
+	component string
+	// generic context options common to all commands
+	*genericclioptions.Context
+}
+
+// NewDescribeComponentOptions creates a new DescribeComponentOptions instance
+func NewDescribeComponentOptions() *DescribeComponentOptions {
+	return &DescribeComponentOptions{}
+}
+
+// Complete completes DescribeComponentOptions after they've been created
+func (o *DescribeComponentOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
```

Comment:
You should use ConcurrentTask similarly to what's done for `catalog list components` to fetch list in parallel. For that matter, it might make sense to investigate whether some of code could be shared between the catalog commands since there is a lot of commonalities.

Reply:
I have added ConcurrentTask to this function, however, I struggled with abstracting the code between the catalog commands. I couldn't figure out how to pass some args (e.g. `o.Client`) to the abstracted function and how to return the values of a ConcurrentTasks in that same function :(

Reply:
No problem. I'll take a look once this is merged.